compiler/rustc_parse/messages.ftl

```diff
@@ -14,10 +14,6 @@ parse_array_index_offset_of = array indexing not supported in offset_of
 
 parse_assignment_else_not_allowed = <assignment> ... else {"{"} ... {"}"} is not allowed
 
-parse_assoc_lifetime = associated lifetimes are not supported
```

Comment:
this message was confusing imo, it talked about a “fictive” Rust feature, *associated lifetimes* as if that term was common knowledge. further, I would expect this message to trigger on `X<'a = 'static>` (`'a` is the “associated lifetime”, `'static` is the *projected* lifetime), not on `X<T = 'static>` as it currently does.